Design/Rule0016CheckForMissingCaptions.cs

```diff
@@ -26,6 +27,12 @@ private void CheckForMissingCaptions(SymbolAnalysisContext context)
             if (context.Symbol.IsObsoletePending || context.Symbol.IsObsoleteRemoved) return;
             if (context.Symbol.GetContainingObjectTypeSymbol().IsObsoletePending || context.Symbol.GetContainingObjectTypeSymbol().IsObsoleteRemoved) return;
 
+            LinterSettings.Create();
+            if (!(LinterSettings.instance.enableRule0016ForApiObjects) && context.Symbol.GetContainingObjectTypeSymbol().GetProperty(PropertyKind.PageType) != null)
+            {
+                if (context.Symbol.GetContainingObjectTypeSymbol().GetProperty(PropertyKind.PageType).ValueText == PageTypeKind.API.ToString()) return;
```

Comment:
Do you really need to compare them as text here? I think it should be fine without `ValueText` and without `ToString()`

Reply:
@StefanMaron, thank you for reviewing this and your feedback.

I'm not sure howto handle the comparison directly between the PageType and the PageTypeKind. Can I directly do and compare between them?

Reply:
I think you could first cast the output from `GetContainingObjectTypeSymbol()` to the page symbol class and then just use the `.PageType` property of that object (which returns a `PageTypeKind` enum value.

I would really love to see this PR merged and released 😊 as this is a very annoying false positive.

Reply:
Hi @StefanMaron, I've (finally) had the time to rework this. Again, thank you for reviewing this, and you were right that the String/Value wasn't the best solution.